cfme/middleware/deployment.py

```diff
@@ -43,91 +56,103 @@ class MiddlewareDeployment(MiddlewareBase):
     Args:
         name: Name of the deployment
         provider: Provider object (HawkularProvider)
-        id: Native id (internal id) of deployment
         server: Server name of the deployment
+        id: Native id (internal id) of deployment
 
     Usage:
 
         mydeployment = MiddlewareDeployment(name='Foo.war',
-                                server='Bar-serv',
+                                server=ser_instance,
                                 provider=haw_provider)
 
-        deployments = MiddlewareDeployment.deployments()
+        deployments = MiddlewareDeployment.deployments() [or]
+        deployments = MiddlewareDeployment.deployments(provider=haw_provider) [or]
+        deployments = MiddlewareDeployment.deployments(provider=haw_provider,server=ser_instance)
 
     """
     property_tuples = [('name', 'name')]
 
-    def __init__(self, name, provider=None, **kwargs):
+    def __init__(self, name, provider=None, server=None, **kwargs):
         if name is None:
             raise KeyError("'name' should not be 'None'")
+        if server is None or not isinstance(server, MiddlewareServer):
```

Comment:
can server be None?
if yes then I think check must be:
if server and not isinstance(server, MiddlewareServer):